sageresearch-app-sdk/src/test/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepositoryTest.kt

```diff
@@ -0,0 +1,397 @@
+
+        // now that failure has come back for the batch
+        testSchedulerRule.testScheduler.advanceTimeBy(1, SECONDS)
+
+        // we should get call updateScheduleToBridge once for each entity
+        // in real practice, at least one of them should result in an unrecoverable failure
+        verify<ScheduleRepository>(scheduleRepository).updateScheduleToBridge(scheduledActivityEntity1)
+        verify<ScheduleRepository>(scheduleRepository).updateScheduleToBridge(scheduledActivityEntity2)
+        verifyNoMoreInteractions(scheduleRepository)
+        inOrder.verifyNoMoreInteractions()
+    }
+
+    @Test
+    fun updateScheduleToBridge() {
+        val scheduledActivity = mock<ScheduledActivity>()
+        val scheduledActivityEntity = mock<ScheduledActivityEntity>()
+        `when`(scheduledActivityEntity.clientWritableCopy()).thenReturn(scheduledActivity)
+
+        `when`(activityManager.updateActivities(eq(listOf(scheduledActivity))))
+                .thenReturn(toV1Single(Single.just(Message()).delay(1, SECONDS)))
+
+        val testObserver = scheduleRepository.updateScheduleToBridge(scheduledActivityEntity).test()
+
+        verify(activityManager).updateActivities(eq(listOf(scheduledActivity)))
+
+        val inOrder = inOrder(scheduledActivityEntity, scheduledActivityEntityDao)
+
+        inOrder.verify(scheduledActivityEntity)
+                .needsSyncedToBridge = true
+        inOrder.verify<ScheduledActivityEntityDao>(scheduledActivityEntityDao)
+                .upsert(eq(listOf(scheduledActivityEntity)))
+        inOrder.verifyNoMoreInteractions()
+
+        // expect
+        testObserver.assertNotComplete()
+
+        // return from the response
+        testSchedulerRule.testScheduler.advanceTimeBy(1, SECONDS)
+
+        inOrder.verify(scheduledActivityEntity)
+                .needsSyncedToBridge = false
+        inOrder.verify<ScheduledActivityEntityDao>(scheduledActivityEntityDao)
+                .upsert(eq(listOf(scheduledActivityEntity)))
+        inOrder.verifyNoMoreInteractions()
+
+        testObserver.assertComplete()
+    }
+
+    @Test
+    fun updateScheduleToBridge_unrecoverable() {
+        val scheduledActivity = mock<ScheduledActivity>()
+        val scheduledActivityEntity = mock<ScheduledActivityEntity>()
+
+        `when`(scheduledActivityEntity.clientWritableCopy()).thenReturn(scheduledActivity)
+
+        // irrecoverable response from Bridge
+        `when`(activityManager.updateActivities(eq(listOf(scheduledActivity))))
+                .thenReturn(toV1Single(
+                        Single.error<Message>(Throwable("Client data too large")).delay(1, SECONDS)))
+
+        val testObserver = scheduleRepository.updateScheduleToBridge(scheduledActivityEntity).test()
+
+        verify(activityManager).updateActivities(eq(listOf(scheduledActivity)))
+
+        val inOrder = inOrder(scheduledActivityEntity, scheduledActivityEntityDao)
+
+        inOrder.verify(scheduledActivityEntity)

```

Comment:
@dephillipsmichael I made the refactor I proposed so updateScheduleToBridge now

1. marks entity as `needsSyncedToBridge=true` and save to DB
2. PUT schedule to Bridge
3. on receiving response
   - success: marks entity as `needsSyncedToBridge=false` and save to DB 
   - recoverable failure: do nothing
   - unrecoverable failure: marks entity as `needsSyncedToBridge=false` and save to DB